Single submit file

ChromeHearts opened the pull request:
<!-- Thank you for the pull request! -->

#### Reference Issue
#95 

#### What does this pull request implement/fix? Explain your changes.

Added single_file in dagman.build() to build submit files in single_submit_file mode.

#### Any other comments?

I have rebased with the latest master and made sure all unittests passed

jrbourbeau commented:
Thank you for the PR @ChromeHearts! I'm currently at a conference, but will review this PR when I get back. If I don't review after a week, please feel free to bug me : )

ChromeHearts commented:
@jrbourbeau I have rebased my PR with the latest master and added unites for the single_submit_file dag.  Please help review and merge it.  I made sure it passed all unittests as well. Thanks!

jrbourbeau commented on pycondor/dagman.py:
```diff
@@ -289,11 +333,31 @@ def build(self, makedirs=True, fancyname=True):
+
+            if not os.path.exists(job_submit_file):
+                # Write lines to the single job submit file
+                with open(job_submit_file, 'w') as file:
```
This is a minor comment, but can you replace `file` here with something else, perhaps `f` or `submit_file`, to avoid confusion with the built-in `file` function in Python 2

jrbourbeau commented on pycondor/dagman.py:
```diff
@@ -31,7 +46,10 @@ def _iter_job_args(job):
-        prior to using _iter_job_args.
+        prior to using _iter_job_args in non-single_file mode
+
+    single_file : single_file
```
See my other comment about the `single_file` description

jrbourbeau commented on pycondor/dagman.py:
```diff
@@ -404,12 +468,15 @@ def build_submit(self, makedirs=True, fancyname=True, submit_options=None):
             for possible options).
 
+        single_file: bool, optional
+            When it's true, single job submission file will be generated.  Default is False.
```
Minor comment. Can you update the `single_file` description to describe what happens when `single_file=True` and when `single_file=False`. Perhaps something like:

```suggestion
            Option to create a single ``Job`` submit file. If ``True``, a single ``Job`` submit file will be created and variables for each ``Job`` will be defined in the corresponding ``Dagman`` submit file. If ``False``, then a submit file is generated for each ``Job`` being managed by a ``Dagman`` (default is ``False``).
```

jrbourbeau commented on pycondor/tests/test_dagman.py:
```diff
@@ -293,3 +296,29 @@ def test_dagman_add_node_ignores_duplicates(tmpdir, dagman):
     assert dagman.nodes == [job]
+
+
+def test_dagman_build_single_file(tmpdir):
```
This test should be expanded to ensure that all `Job` submit file attributes (e.g. `queue`, `log`, etc.) are included when `single_file=True`. Using the `pytest.mark.parametrize` decorator could be useful here.
